wgpu-core/src/command/memory_init.rs

```diff
@@ -285,73 +261,36 @@ impl<A: hal::Api> BakedCommands<A> {
                     }
                 }
                 MemoryInitKind::NeedsInitializedMemory => {
-                    ranges.clear();
                     for (mip_level, mip_tracker) in affected_mip_trackers {
-                        for layer_range in mip_tracker.drain(use_range.layer_range.clone()) {
-                            ranges.push(TextureInitRange {
-                                mip_range: mip_level as u32..(mip_level as u32 + 1),
+                        ranges.extend(mip_tracker.drain(use_range.layer_range.clone()).map(
```

Comment:
Extensive use of functional chaining, like we are doing here, becomes an obstruction to reading and debugging the code. In this case, `extend()` doesn't really give us anything practical. But stepping through the loop would be nice.